src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -327,6 +327,7 @@ public Completable start() {
         handshakeResponseSubject = CompletableSubject.create();
         handshakeReceived = false;
         CompletableSubject tokenCompletable = CompletableSubject.create();
+        localHeaders.put("User-Agent", "Microsoft/SignalR/3.1/OtherUserAgentThings");
```

Comment:
Well *that's* definitely not going to be the user agent ;P.

Reply:
Perhaps this? Where `version` is a value get from a constant generated by the build or something.

```suggestion
        localHeaders.put("User-Agent", "Microsoft SignalR/" + version + " (Java)");
```

Reply:
> Well that's definitely not going to be the user agent ;P

I remember hearing stories of `ICanHas` prefixes lol. But yeah, I meant to comment to ask for what we wanted here

Reply:
Oh I ended up with comments. I ended up with more comments that I originally thought (see above) ;P.